### PR TITLE
Issue #13345: Enable examples tests for OverloadMethodsDeclarationOrd…

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  OverloadMethodsDeclarationOrderCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **OverloadMethodsDeclarationOrderCheck** in the project. The implementation focuses on ensuring that the OverloadMethodsDeclarationOrder checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example2.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="OverloadMethodsDeclarationOrder"/>
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example2.java:4]: example with same config look weird in comparison with others.

lets make example1
```
class Example1 {
  void same(int i) {}
  // comments between overloaded methods are allowed.
  void same(String s) {}
  void same(String s, int i) {}
  void same(int i, String s) {}
  void notSame() {}
  interface notSame{}

  void otherSame(String s) {}
  void foo() {}
  // violation below
  void otherSame(String s, int i) {}
}
```

and remove Example2.


change description to `Example of correct and incorrect grouping of overloaded methods` to

**Review by romani — CHANGES_REQUESTED** [review]
item to improve:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example2.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example2.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot.

**Pull request merged**